exam-3-trial-3/src/main/java/app/ConsoleMenu.java

```diff
@@ -0,0 +1,94 @@
+                            "2 - List all animals\n" +
+                            "3 - Find animal by ID\n" +
+                            "4 - Delete animal by ID\n" +
+                            "5 - Record a new adoption\n" +
+                            "6 - List all adoptions\n" +
+                            "7 - Find adoption by ID\n" +
+                            "8 - Delete adoption by ID\n" +
+                            "9 - Exit\n" +
+                            "Please choose an option:\n"
+            );
+            int input = scanner.nextInt();
+            scanner.nextLine();
+            switch (input) {
+                case 1 -> {
+                    System.out.println("Enter animal name: ");
+                    String name = scanner.nextLine();
+                    System.out.println("Enter animal species: ");
+                    String species = scanner.nextLine();
+                    System.out.println("Enter animal age: ");
+                    int age = scanner.nextInt();
+
+                    Animal animal = Animal.builder()
+                            .name(name)
+                            .species(species)
+                            .age(age).build();
+                    scanner.nextLine();
+
+                    animalService.addNewAnimal(animal);
+                }
+                case 2 -> animalService.listAllAnimals().forEach(System.out::println);
+                case 3 -> {
+                    System.out.println("Enter an animal id:");
+                    System.out.println(animalService.getAnimalById(scanner.nextLong()));
+                }
+                case 4 -> {
+                    System.out.println("Enter an animal id to delete:");
+                    animalService.deleteAnimal(scanner.nextLong());
+                }
+                case 5 -> {
+                    System.out.println("Enter an animal id:");
+                    long animalId = scanner.nextLong();
+                    scanner.nextLine();
+                    System.out.println("Enter an adoption name: ");
+                    String adoptionName = scanner.nextLine();
+                    adoptionService.createAdoption(animalId, adoptionName);
+                }
+                case 6 -> adoptionService.listAllAdoption().forEach(System.out::println);
+                case 7 -> {
+                    System.out.println("Enter an adoption id:");
+                    System.out.println(adoptionService.getAdoptionById(scanner.nextLong()));
+                    scanner.nextLine();
+                }
+                case 8 -> {
+                    System.out.println("Enter an adoption id to delete:");
+                    adoptionService.deleteAdoption(scanner.nextLong());
+                }
+                case 9 -> {
+                    scanner.close();
+                    System.exit(0);
+                }
+                default -> System.out.println("Invalid choice");
+            }
```

Comment:
this could be extracted in a method - separation of concerns is important  on this lvl too